### PR TITLE
Va 111 participant who joins before hearing officer or reporter sees a waiting for host message

### PR DESCRIPTION
**Pull request by giladadon1997**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### JIRA link(s):

- [AHOYAPPS-0000](https://issues.corp.twilio.com/browse/AHOYAPPS-0000)

### Description

A description of what this PR does.

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### Before merge
* [ ] Got one or more +1s
* [ ] Re-tested if necessary

### CODE REVIEW

File: src/components/MenuBar/MenuBar.tsx [hunk 1]
```diff
@@ -81,38 +84,51 @@ export default function MenuBar() {
...
-      'District Representative',
-      'Interpreter',
-      'Other',
+      PARTICIANT_TYPE.HEARING_OFFICER,
+      PARTICIANT_TYPE.PARENT,
+      PARTICIANT_TYPE.PARENT_REPRESENTATIVE,
+      PARTICIANT_TYPE.DISTRICT_REPRESENTATIVE,
+      PARTICIANT_TYPE.INTERPRETER,
+      PARTICIANT_TYPE.OTHER,
     ];
```
Comment (talr) [inline src/components/MenuBar/MenuBar.tsx]: I think we should move it to separate file and in the future it should be retrieved from the database.

File: src/components/MenuBar/MenuBar.tsx [hunk 2]
```diff
@@ -81,38 +84,51 @@ export default function MenuBar() {
   const reporterToken: string = window.location.hash.substr(1);
   let repoterInfo: any = reporterToken ? jwt_decode(reporterToken) : '';
   const classes = useStyles();
-
-  const { getToken, isFetching } = useAppState();
+  const { setError, getToken, isFetching } = useAppState();
```
Comment (YehudaNewmanEScribers) [inline src/components/MenuBar/MenuBar.tsx:91]: please see if we have a way to display messages without using setError e.g. something like setInfo

File: src/components/VideoProvider/useRoom/useRoom.tsx [hunk 3]
```diff
@@ -27,7 +28,7 @@ export default function useRoom(localTracks:any, onError: Callback, options?: Co
           setRoom(newRoom);
           const disconnect = () => newRoom.disconnect();
 
-          newRoom.once('disconnected', () => {
+          newRoom.once(ROOMSTATE.DISCONNECTED, () => {
```
Comment (talr) [inline src/components/VideoProvider/useRoom/useRoom.tsx]: I think ROOMSTATE should be two words per our Naming Convention document.

File: src/state/index.tsx [hunk 4]
```diff
@@ -118,6 +116,10 @@ export default function AppStateProvider(props: React.PropsWithChildren<{}>) {
   );
 }
 
+function participantNotHearingOfficerOrReporter(partyType: string) {
```
Comment (talr) [inline src/state/index.tsx]: Please change the function name to more generic name, I would change the logic to opposite i.e. participantIsMemberInHostRole

File: .gitignore [hunk 5]
```diff
@@ -7,7 +7,6 @@
 node_modules
 .pnp
 .pnp.js
-
```
Comment (YehudaNewmanEScribers) [inline .gitignore:10]: please add .vs

File: .gitignore [hunk 6]
```diff
@@ -7,7 +7,6 @@
...
 # testing
 coverage
```
Comment (talr) [inline .gitignore:13]: Please make sure we ignore .vs/slnx.sqlite

**Review by talr — CHANGES_REQUESTED** [review]

File: public/config.json [hunk 7]
```diff
@@ -1,4 +1,4 @@
 {
 	"stage": "dev",
-	"endPoint": "https://digu74hvnk.execute-api.us-east-1.amazonaws.com/dev"
+	"endPoint": "https://digu74hvnk.execute-api.us-east-1.amazonaws.com/dev/"
```
Comment (talr) [inline public/config.json]: Please change the name of the file to config_mt_intg_01_aws_us_east_1.json per our Environments documents.
In the future, we'd like to change this to eScribers.net domain with the exact environment name as part of the URL.
What do we need the stage for?
Shall we change it or add "environment": "mt-intg-01-aws-us-east-1"?

File: src/App.tsx [hunk 8]
```diff
@@ -6,6 +6,7 @@ import LocalVideoPreview from './components/LocalVideoPreview/LocalVideoPreview'
 import MenuBar from './components/MenuBar/MenuBar';
 import ReconnectingNotification from './components/ReconnectingNotification/ReconnectingNotification';
 import Room from './components/Room/Room';
+import { ROOMSTATE } from 'utils/displayStrings';
```
Comment (talr) [inline src/App.tsx]: I think ROOMSTATE is two words.

File: src/__mocks__/twilio-video.ts [hunk 9]
```diff
@@ -1,4 +1,5 @@
 import { EventEmitter } from 'events';
+import { TRACK_TYPE } from '../utils/displayStrings';
 
 class MockRoom extends EventEmitter {
```
Comment (talr) [inline src/__mocks__/twilio-video.ts:4]: Let's discuss the concept behind MockRoom? Maybe a comment / documentation here will help.

File: src/components/ErrorDialog/enhanceMessage.ts [hunk 10]
```diff
@@ -1,9 +1,10 @@
+import { EROOR_MESSAGE } from '../../utils/displayStrings';
 // This function is used to provide error messages to the user that are
 // different than the error messages provided by the SDK.
 export default function enhanceMessage(message = '', code?: number) {
   switch (code) {
     case 20101: // Invalid token error
-      return message + '. Please make sure you are using the correct credentials.';
+      return message + EROOR_MESSAGE.INVALIDE_CREDENTIALS;
```
Comment (talr) [inline src/components/ErrorDialog/enhanceMessage.ts]: I think we should change INVALIDE to INVALID

File: src/components/MenuBar/MenuBar.tsx [hunk 11]
```diff
@@ -11,16 +11,18 @@ import Toolbar from '@material-ui/core/Toolbar';
...
 import LocalAudioLevelIndicator from './LocalAudioLevelIndicator/LocalAudioLevelIndicator';
 import ToggleFullscreenButton from './ToggleFullScreenButton/ToggleFullScreenButton';
 import ToggleGridViewButton from './ToggleGridViewButton/ToggleGridViewButton';
 import SettingsButton from './SettingsButton/SettingsButton';
-
 import { useAppState } from '../../state';
 import useRoomState from '../../hooks/useRoomState/useRoomState';
 import useVideoContext from '../../hooks/useVideoContext/useVideoContext';
 
+const JOIN_ROOM = 'Join Room';
```
Comment (talr) [inline src/components/MenuBar/MenuBar.tsx]: If this is a message, maybe we should add suffix such as _MESSAGE or _LABEL.
Can we make all messages in associative array? i.e. JOIN_ROOM_MESSAGE['en-US'] = 'Join Room';
Can we move all those _MESSAGES / _LABELS to separate resource file (i.e. displayStrings.ts)?

File: src/utils/displayStrings.ts [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+export const PLAYER_STATE = {
+  stopped: 'stopped',
+};
+
+export const ROOMSTATE = {
+  RECONNECTING: 'reconnecting',
+  RE_CONNECTED: 'reconnected',
+  DISCONNECTED: 'disconnected',
+  CONNECTED: 'connected',
```
Comment (talr) [inline src/utils/displayStrings.ts]: Are all those displayStrings or some of them are for internal usage? If so, I would separate them to two files, with different object / variable suffix.

File: src/utils/displayStrings.ts [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+  NETWORK_ERROR: 'Network Error',
+};
+
+export const PLAYER_STATE = {
+  stopped: 'stopped',
+};
+
+export const ROOMSTATE = {
+  RECONNECTING: 'reconnecting',
+  RE_CONNECTED: 'reconnected',
```
Comment (talr) [inline src/utils/displayStrings.ts]: I think RE_CONNECTED is one word.

**Review by talr — CHANGES_REQUESTED** [review]
Can we have unit tests, system tests, e2e tests to the new functionality?

File: .gitignore [hunk 14]
```diff
@@ -7,7 +7,9 @@
 node_modules
 .pnp
 .pnp.js
-
+.vs
```
Comment (YehudaNewmanEScribers) [inline .gitignore]: should be .vs/*

File: .gitignore [hunk 15]
```diff
@@ -7,7 +7,9 @@
...
+.sqlite
```
Comment (YehudaNewmanEScribers) [inline .gitignore]: you can take this out because of the .vs folder. and also the next line

File: src/__mocks__/twilio-video.ts (hunk 9 above)
Comment (YehudaNewmanEScribers) [reply src/__mocks__/twilio-video.ts:4]: Needs to be familiarised as we continue to familiarise ourselves with the code base

File: src/components/ErrorDialog/enhanceMessage.ts [hunk 16]
```diff
@@ -1,9 +1,10 @@
+import { EROOR_MESSAGE } from '../../utils/displayStrings';
 // This function is used to provide error messages to the user that are
 // different than the error messages provided by the SDK.
 export default function enhanceMessage(message = '', code?: number) {
   switch (code) {
     case 20101: // Invalid token error
-      return message + '. Please make sure you are using the correct credentials.';
+      return message + EROOR_MESSAGE.INVALID_CREDENTIALS;
```
Comment (YehudaNewmanEScribers) [inline src/components/ErrorDialog/enhanceMessage.ts]: The word error is spelled 'ERROR'. Please change in all places

File: src/state/index.tsx [hunk 17]
```diff
@@ -34,7 +36,7 @@ export default function AppStateProvider(props: React.PropsWithChildren<{}>) {
   const reporterToken = window.location.hash.substr(1);
 
   var endpoint = '';
-  fetch(`${process.env.PUBLIC_URL}/config.json`)
+  fetch(`${process.env.PUBLIC_URL}/config_mt_intg_01_aws_us_east_1.json`)
```
Comment (YehudaNewmanEScribers) [inline src/state/index.tsx]: This still needs to be config.json even though the file doesn't exist in the repository.
Please update the readme file (or create it if it doesn't exist) to inform developers that they need to create the public/config.json file from one of the examples there depending on the environment they wish to test against (and to possibly update the value of the url if it has been changed by the CICD)

**Review by YehudaNewmanEScribers — CHANGES_REQUESTED** [review]

File: public/config_json_example/config_mt_intg_01_aws_us_east_1.json [hunk 18]
```diff
@@ -0,0 +1,4 @@
+{
+     "endPoint":"https://digu74hvnk.execute-api.us-east-1.amazonaws.com/dev",
+      "stage":"dev"
```
Comment (talr) [inline public/config_json_example/config_mt_intg_01_aws_us_east_1.json]: Does Twilio use this? Do we use it?

**Review by talr — CHANGES_REQUESTED** [review]

File: src/components/MenuBar/MenuBar.tsx (hunk 11 above)
Comment (talr) [reply src/components/MenuBar/MenuBar.tsx]: ?

**Review by talr — APPROVED** [review]

**Review by YehudaNewmanEScribers — APPROVED** [review]

**Pull request merged**